File: 2022/day-02/MathyouMB/day2.rs
use std::fs::File;
use std::io::prelude::*;
use std::collections::HashMap;

fn main() {
    let mut file = File::open("input.txt")
        .expect("File not found");

    let mut data = String::new();

    file.read_to_string(&mut data)
        .expect("Error while reading file");

    let opponent = HashMap::from([
        ("A", 1),
        ("B", 2),
        ("C", 3)
    ]);
    
    let player = HashMap::from([
        ("X", 1),
        ("Y", 2),
        ("Z", 3)
    ]);

    // problem 1
    let mut score = 0;
    for line in data.lines() {
        let mut iter = line.split_whitespace();
        let a = iter.next().unwrap();
        let b = iter.next().unwrap();

        if opponent[a] == player[b] {
            score += 3;
        } else if (a == "A" && b == "Y") || (a == "B" && b == "Z") || (a == "C" && b == "X"){
            score += 6;
        }
        score += player[b];
    }

    // problem
    let mut score2 = 0;
    for line in data.lines() {
        let mut iter = line.split_whitespace();
        let a = iter.next().unwrap();
        let b = iter.next().unwrap();

        if b == "X" {
            score2 += 0;
            if a == "A" {
                score2 += player["Z"];
            } else if a == "B" {
                score2 += player["X"];
            } else {
                score2 += player["Y"];
            }
        } else if b == "Y" {
            score2 += 3;
            if a == "A" {
                score2 += player["X"];
            } else if a == "B" {
                score2 += player["Y"];
            } else {
                score2 += player["Z"];
            }
        } else {
            score2 += 6;
            if a == "A" {
                score2 += player["Y"];
            } else if a == "B" {
                score2 += player["Z"];
            } else {
                score2 += player["X"];
            }
        }
    }

    println!("P1: {}", score);
    println!("P2: {}", score2);
}

File: 2022/day-02/victorli5611/day2.py
data = [l.strip() for l in open('day2.txt')]
p1 = 0
p2 = 0
for line in data:
    op,me = line.split()
    p1 += {'X': 1, 'Y': 2, 'Z': 3}[me]
    p1 += {('A', 'X'): 3, ('A', 'Y'): 6, ('A', 'Z'): 0,
            ('B', 'X'): 0, ('B', 'Y'): 3, ('B', 'Z'): 6,
            ('C', 'X'): 6, ('C', 'Y'): 0, ('C', 'Z'): 3,
            }[(op, me)]

    p2 += {'X': 0, 'Y': 3, 'Z': 6}[me]
    p2 += {('A', 'X'): 3, ('A', 'Y'): 1, ('A', 'Z'): 2,
            ('B', 'X'): 1, ('B', 'Y'): 2, ('B', 'Z'): 3,
            ('C', 'X'): 2, ('C', 'Y'): 3, ('C', 'Z'): 1,
            }[(op, me)]
print(p1)
print(p2)
